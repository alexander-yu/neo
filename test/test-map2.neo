int multiply(int i, int j) {
    return i * j;
}

float add(float i, float j) {
    return i + j;
}

void main() {
    create matrix<int> m1 = [[1,2],[2,3]];
    create matrix<int> m2 = [[4,3],[4,1]];
    println(imap2(multiply, m1, m2));
    create matrix<float> m3 = [[1.1,4.3,2.8],[4.7,6.2,3.0]];
    create matrix<float> m4 = [[3.2,2.5,4.0],[1.9,5.3,2.6]];
    println(fmap2(add, m3, m4));
    create matrix<float> m5 = [[1., 2.], [3., 4.]];
    create matrix<float> m6 = [[1., 2.]];
    println(fmap2(add, m5, m6));
}
