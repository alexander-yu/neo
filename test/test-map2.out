[[4, 6],
 [8, 3]]
[[4.3, 6.8, 6.8],
 [6.6, 11.5, 5.6]]
[[2, 4],
 [4, 6]]
